NocoDB Node updated for v0.90.0 - N8N-3411

mertmit opened the pull request:
- Added an option to use new API for NocoDB. (Old API will be deprecated on 30 Jun 2022)

mertmit committed:
feat(NocoDB Node): add support for new NocoDB API

mertmit committed:
fix(NocoDB Node): fix binary row update on old NocoDB API

mertmit committed:
fix(NocoDB Node): fix getAll endpoint

mertmit committed:
feat(NocoDB Node): allow xc-token as credential

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/n8n-io/n8n?pullRequest=3146) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/n8n-io/n8n?pullRequest=3146) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/n8n-io/n8n?pullRequest=3146) it.</sub>

mertmit committed:
fix(NocoDB Node): get all for new api

mertmit committed:
feat(NocoDB Node): list options & change to single data apis

o1lab commented:
With NocoDB version >0.90 the API docs have changed. @janober - if you could pls see the merge through it will help the community. @mertmit can help follow up with any comments to the PR. Thank you

ed-parsadanyan commented:
Hello. As mentioned in [#3170](https://github.com/n8n-io/n8n/issues/3170) the issue is still here.
From the commit code I would expect new options in the NocoDB Node (see screenshot), but in the current Docker version of n8n (174.0) I don't see any changes in the interface.

![image](https://user-images.githubusercontent.com/10472129/166248638-78289d3a-9202-448b-8895-798494cac95d.png)

ed-parsadanyan commented:
> > but in the current Docker version of n8n (0.174.0)
> 
> This PR is yet to be merged - did you expect to still see this in 0.174.0 ?

Oh, I see now, it wasn't merged yet. Hope it will be completed for the next n8n release.

Joffcom committed:
Merge branch 'master' of github.com:n8n-io/n8n into nocodb

Joffcom commented:
Hey @o1lab & @ed-parsadanyan,

This one has not been reviewed yet it has been on my list for a while but we have been chatting internally about how to deal with API changes like this in a way that can work for existing and future nodes.

I am going to be taking a look today to see how this works out with our new approach.

o1lab commented:
@Joffcom : Awesome  - thank you.

Joffcom committed:
Moved to new format and reduced some code reuse

fishnux commented:
Any progress on this review for merge, @Joffcom? Thank you!

Joffcom commented:
Hey @fishnux,

I was off last week but the good news is we have worked out how to handle API versioning in a way that should scale for all nodes in the future. I will have a few tweaks to make to this which are at the top of my list of things to do so we should be have this mostly wrapped up this week.

Joffcom committed:
Added API Version to Node Settings

Joffcom committed:
Merge branch 'master' of github.com:n8n-io/n8n into nocodb

Joffcom committed:
Improvements to remove code reuse and use bulk endpoints

Joffcom committed:
Added new credentials to close off PR#2909

Joffcom committed:
Credential testing working

RicardoE105 committed:
:zap: Improvements

RicardoE105 committed:
⬆️ Merge master

RicardoE105 committed:
:zap: Add generic authentication type to credentials

RicardoE105 committed:
:fire: Remove credentials verification

Joffcom committed:
Fixed Get All not working with manual limit

Joffcom committed:
Removed json object from project / table fields

Joffcom committed:
Merge branch 'master' of github.com:n8n-io/n8n into nocodb

Joffcom committed:
added fix from n8n-4159

RicardoE105 committed:
:shirt: Fix linting issue

RicardoE105 committed:
Merge branch 'master' into nocodb

janober committed:
feat: Improvements to pairedItem

ivov committed:
refactor: Consolidate hoisted package versions (#3724)

* :package: Consolidate hoisted package versions

* :package: Update `package-lock.json`

* :package: Update `package-lock.json`

* :package: Update `package-lock.json`

ivov committed:
refactor: Upgrade to ESLint 8 (#3722)

* :arrow_up: Upgrade to ESLint 8

* :package: Update package-lock.json

* :shirt: Add lint exceptions

* :shirt: Add more lint exceptions

* :heavy_minus_sign: Remove `tslint` from some packages

* :shirt: Except init file

* :package: Update `package-lock.json`

* :package: Update `package-lock.json`

* :shirt: Add exceptions to new lines coming from `master `

Co-authored-by: Jan Oberhauser <jan.oberhauser@gmail.com>

ivov committed:
refactor: Format all credentials (#3720)

* Apply Prettier to all credentials

* Fix quotes for lint

* :shirt: Remove `quotemark` rule

* :shirt: Run Prettier to take over quotes

* :arrow_up: Upgrade `eslint-plugin-n8n-nodes-base`

* :package: Update `package-lock.json`

Co-authored-by: Omar Ajoue <krynble@gmail.com>
Co-authored-by: Jan Oberhauser <jan.oberhauser@gmail.com>

ahsanv committed:
fix: Fix node_type property in all events (#3759)

janober committed:
:arrow_up: Update package-lock.json file

Joffcom committed:
fix(Mautic Node): Fix authentication issue (#3761)

* Fixes mautic credential issue

* removed unused imports

Co-authored-by: Jan Oberhauser <jan.oberhauser@gmail.com>

drudge committed:
fix(AWS DynamoDB Node): Fix expression attribute names (#3763)

* Fix expression attribute names in getAll

* fix: EAN value should be a string, not object

janober commented on packages/nodes-base/nodes/NocoDB/GenericFunctions.ts:
```diff
@@ -35,14 +35,28 @@ interface IAttachment {
+	const authenticationMethod = this.getNodeParameter('authentication', 0);
+	let credentials;
+
+	if (authenticationMethod === 'nocoDbApiToken') {
+		credentials = await this.getCredentials('nocoDbApiToken');
+	} else if (authenticationMethod === 'nocoDb') {
+		credentials = await this.getCredentials('nocoDb');
+	}
```
Can we not simply do the following here?
`credentials = await this.getCredentials(authenticationMethod);`

Joffcom replied:
Changed 👍🏻

janober commented on packages/nodes-base/nodes/NocoDB/GenericFunctions.ts:
```diff
@@ -55,8 +69,10 @@ export async function apiRequest(this: IHookFunctions | IExecuteFunctions | ILoa
 		delete options.body;
 	}
 
+	const credentialType = authenticationMethod === 'nocoDb' ? 'nocoDb' : 'nocoDbApiToken';
+
```
Is this necessary? Is the value of `credentialType` not simply again the value of `authenticationMethod` afterward?

Joffcom replied:
It is, Good catch. Have now changed it.

Joffcom committed:
Removed extra code for working out what credentials are in use

Joffcom committed:
Merge branch 'nocodb' of github.com:mertmit/n8n into nocodb

mutdmour committed:
fix(editor): Fix linking buttons color (#3770)

* fix color of icon

* center buttons

ivov committed:
fix(editor): Restore pindata header colors (#3758)

* :art: Restore `color-secondary-tint` colors

* :fire: Remove typing from JS file

mutdmour committed:
fix(editor): Fix sticky duplication and position bug (#3755)

* fix bug when inserting sticky

* center sticky on insert

* export as const

mutdmour committed:
refactor(editor): Move all colors to css variables (#3723)

* update white color

* update white color

* update more whites

* update color

* update curr running

* update text color #555

* update white color

* set search bar colors

* update colors

* update node executing

* update text colors

* update light color

* update theme

* update theme

* update overlays carousel

* update theme vars

* add dark theme tokens

* update text

* update table colors

* fix conflict

* update colors

agobrech committed:
feat(Metabase Node): Add Metabase Node (#3033)

* Boilerplate with new node's version for metabse

* Metabases MVP features

* Added new credential for metabse, added custom auth for metabase

* Fixed bug with one enpoint not working

* Clean up code

* Uniformised the renovate token

* Made two example of responses for review

* Fixed lint issues

* Feature add datasources

* Changed output from databases

* Changed questions data output

* Fixed issue when testing credentials with new node format

* Add the possibility to get raw data

* Removed handle for the metabase meta results, changed export's name

* Add binary extraction for the result data

* Fixed binary download issue

* :zap: Add preAuthentication method to credentials

* Revert "Added new credential for metabse, added custom auth for metabase"

This reverts commit 5f1b7607adb85d6ec897b184853bdfdbae77df6d.

* Revert "Added new credential for metabse, added custom auth for metabase"

This reverts commit 5f1b7607adb85d6ec897b184853bdfdbae77df6d.

* Added preAuth and fixed autfixable linting rules

* Fixed linting errors

* Linting fixes

* Remove / at the end of url, and add placeholder for cred url

* Make export to Json retun only json and no binary

* Fix lint issues

* Add action and exception for lint rule

* Remove unnecessary credential file

* :zap: Simplify and cleanup

Co-authored-by: ricardo <ricardoespinoza105@gmail.com>
Co-authored-by: Omar Ajoue <krynble@gmail.com>
Co-authored-by: Jan Oberhauser <jan.oberhauser@gmail.com>

mutdmour committed:
fix(editor): Fix spaces bug (#3774)

maxtkacz committed:
refactor(editor): Change welcome sticky content (#3769)

* Updated Welcome sticky content

* Updated welcome sticky thumbnail image

Image was swapped out, used exact same file name + sizing so did not require code changes.

* Replaced welcome sticky thumbnail image

Joffcom committed:
fix(Fix Rocketchat Node): Fix authentication issue (#3778)

ivov committed:
Add suggested VSCode settings (#3783)

* :zap: Add suggested settings

* :fire: Remove app-level setting

* :art: Update indentation

netroy committed:
fix(core): Add windows support to import:credentials --separate (#3589)

michael-radency committed:
feat(Item List Node): Add operation for creating array from input items (#3149)

* :hammer: create array operation

* :hammer: removed semicolumn

* :hammer: updated UI

* :zap: display option fix

* :zap: aggregate operation description update, default aggregate item

MiloradFilipovic committed:
refactor: Add Onboarding call prompts (#3682)

* ✨ Implemented initial onboarding call prompt logic

* ✨ Added onboarding call prompt feature environment variable

* ✨ Implemented onboarding session signup modal

* 📈 Added initial telemetry for the onboarding call prompt

* ✔️ Fixing linter error in server.ts

* 💄 Updating onboaring call prompt and modal wording and styling

* ✨ Implemented initial version of fake doors feature

* ✨ Added parameters to onboarding call prompt request

* ✨ Finished implementing fake doors in settings

* 🔨 Updating onboarding call prompt fetching logic (fetching before timeout starts)

* 👌 Updating onboarding call prompt and fake door components based on the front-end review feedback

* ✨ Updated fake doors so they support UI location specification. Added credentials UI fake doors.

* ⚡ Added checkbox to the signup form, improved N8NCheckbox formatting to better handle overflow

* 💄 Moving seignup checkbox label text to i18n file, updating checkbox component css to force text wrap

* ✨ Update API calls to work with the new workflow request and response formats

* 👌 Updating fake door front-end based on the review feedback

* 👌 Updating onboarding call prompt and fake doors UI based in the product feedback

* ✨  Updated onboarding call prompts front-end to work with new endpoints and added new telemetry events

* 🐛 Fixing onboarding call prompts not appearing in first user sessions

* ⚡️ add createdAt to PublicUser

* 👌 Updating onboarding call prompts front-end to work with the latest back-end and addressing latest product review

* ✨ Improving error handling when submitting user emails on signup

* 💄 Updating info text on Logging feature page

* 💄 Updating first onboarding call prompt timeout to 5 minutes

* 💄 Fixing `N8nCheckbox` component font overflow

Co-authored-by: Ben Hesseldieck <b.hesseldieck@gmail.com>

michael-radency committed:
feat(Kafka Trigger Node): Add additional options (#3600)

* :hammer: additional options to kafka trigger

* :zap: option for maxInFlightRequests

* :zap: Small change

Co-authored-by: ricardo <ricardoespinoza105@gmail.com>

alexgrozav committed:
fix(editor): Fix pin data in executions when pinData is null. (#3787)

janober committed:
:arrow_up: Update package-lock.json file

janober committed:
:bookmark: Release n8n-workflow@0.110.0

janober committed:
:bookmark: Release n8n-core@0.128.0

janober committed:
:arrow_up: Set n8n-workflow@0.110.0 on n8n-core

janober committed:
:bookmark: Release n8n-node-dev@0.67.0

janober committed:
:arrow_up: Set n8n-core@0.128.0 and n8n-workflow@0.110.0 on n8n-node-dev

janober committed:
:arrow_up: Set n8n-core@0.128.0 and n8n-workflow@0.110.0 on n8n-nodes-base

janober committed:
:bookmark: Release n8n-nodes-base@0.186.0

janober committed:
:bookmark: Release n8n-design-system@0.28.0

janober committed:
:bookmark: Release n8n-editor-ui@0.154.0

janober committed:
:arrow_up: Set n8n-design-system@0.28.0 and n8n-workflow@0.110.0 on n8n-editor-ui

janober committed:
:arrow_up: Set n8n-core@0.128.0, n8n-editor-ui@0.154.0, n8n-nodes-base@0.186.0 and n8n-workflow@0.110.0 on n8n

janober committed:
:bookmark: Release n8n@0.188.0

janober committed:
:bookmark: Update main package.json to 0.188.0

janober committed:
:books: Update CHANGELOG.md with version 0.188.0

ivov committed:
:shirt: Adjust line endings for Prettier lintings (#3786)

janober committed:
build: Use package-lock.json file with custom build

MiloradFilipovic committed:
💄 Updating onboarding prompt label

ivov committed:
:arrow_up: Set eslint@8.0.0 on n8n-workflow (#3768)

* :arrow_up: Upgrade `n8n-workflow` to ESLint 8

* :package: Update `package-lock.json`

* :package: Re-update `package-lock.json`

MiloradFilipovic committed:
Merge pull request #3801 from n8n-io/onboarding-call-prompt-label-update

💄 Updating onboarding prompt label

janober committed:
:twisted_rightwards_arrows: Merge branch 'nocodb' of github.com:mertmit/n8n into mertmit-nocodb

janober committed:
:zap: Fix on error behaviour for Delete, Get and Update

janober commented:
Thanks a lot. Got merged.

janober commented:
Got released with `n8n@0.189.0`
